point_viewer_grpc/src/bin/octree_benchmark.rs

```diff
@@ -58,38 +66,60 @@ fn main() {
     );
     let num_points = usize::from_str(matches.value_of("num-points").unwrap_or("50000000"))
         .expect("num-points needs to be a number");
+    let num_threads = usize::from_str(
+        matches
+            .value_of("num-threads")
+            .unwrap_or(&(num_cpus::get() - 1).to_string()),
+    )
+    .expect("num-threads needs to be a number");
+    let buffer_size = usize::from_str(matches.value_of("buffer-size").unwrap_or("4"))
+        .expect("buffer-size needs to be a number");
     if matches.is_present("no-client") {
-        server_benchmark(&octree_directory, num_points)
+        server_benchmark(&octree_directory, num_points, num_threads, buffer_size)
     } else {
         let port = value_t!(matches, "port", u16).unwrap_or(50051);
         full_benchmark(&octree_directory, num_points, port)
     }
 }
 
-fn server_benchmark(octree_directory: &Path, num_points: usize) {
-    let octree = octree_from_directory(octree_directory).unwrap_or_else(|_| {
-        panic!(
-            "Could not create octree from '{}'",
-            octree_directory.display()
-        )
-    });
+fn server_benchmark(
+    octree_directory: &Path,
+    num_points: usize,
+    num_threads: usize,
+    buffer_size: usize,
+) {
+    let octree: [Octree; 1] = [
+        *octree_from_directory(octree_directory).unwrap_or_else(|_| {
+            panic!(
+                "Could not create octree from '{}'",
+                octree_directory.display()
+            )
+        }),
+    ];
     let mut counter: usize = 0;
+    let mut points_streamed_m = 0;
     let all_points = PointQuery {
         location: PointLocation::AllPoints(),
         global_from_local: None,
     };
-    let mut batch_iterator = BatchIterator::new(&octree, &all_points, BATCH_SIZE);
-
+    let mut batch_iterator =
+        BatchIterator::new(&octree, &all_points, BATCH_SIZE, num_threads, buffer_size);
+    println!("Server benchmark:");
     let _result = batch_iterator.try_for_each_batch(move |point_data| {
         counter += point_data.position.len();
+
+        if points_streamed_m < counter / BATCH_SIZE {
```

Comment:
How about replacing `BATCH_SIZE` here and in the next line by `1_000_000`? We may change `BATCH_SIZE` to another number in the future...

Reply:
that makes sense